File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/types/SearchResultItem.java
package com.ucotron.sdk.types;

/**
 * A single search result with relevance scores.
 */
public class SearchResultItem {
    private long id;
    private String content;
    private String node_type;
    private float score;
    private float vector_sim;
    private float graph_centrality;
    private float recency;

    public long getId() { return id; }
    public String getContent() { return content; }
    public String getNodeType() { return node_type; }
    public float getScore() { return score; }
    public float getVectorSim() { return vector_sim; }
    public float getGraphCentrality() { return graph_centrality; }
    public float getRecency() { return recency; }
}

File: php/src/Types/SearchResultItem.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Types;

class SearchResultItem
{
    public function __construct(
        public readonly string $id,
        public readonly string $content,
        public readonly float $score,
        public readonly ?string $node_type = null,
        public readonly ?int $timestamp = null,
        public readonly ?array $metadata = null,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            content: $data['content'] ?? '',
            score: (float) ($data['score'] ?? 0.0),
            node_type: $data['node_type'] ?? null,
            timestamp: isset($data['timestamp']) ? (int) $data['timestamp'] : null,
            metadata: $data['metadata'] ?? null,
        );
    }
}
